src/internal_events/aws_s3.rs

```diff
@@ -0,0 +1,119 @@
+        fn emit_logs(&self) {
+            warn!(message = "Failed to fetch SQS events.", %self.error);
+        }
+
+        fn emit_metrics(&self) {
+            counter!("sqs_message_receive_failed_total", 1);
+        }
+    }
+
+    #[derive(Debug)]
+    pub(crate) struct SqsMessageReceiveSucceeded {
+        pub count: usize,
+    }
+
+    impl InternalEvent for SqsMessageReceiveSucceeded {
+        fn emit_logs(&self) {
+            trace!(message = "Received SQS messages.", %self.count);
+        }
+
+        fn emit_metrics(&self) {
+            counter!("sqs_message_receive_succeeded_total", 1);
+            counter!("sqs_message_received_messages", self.count as u64,);
+        }
+    }
+
+    #[derive(Debug)]
+    pub(crate) struct SqsMessageProcessingSucceeded<'a> {
+        pub message_id: &'a str,
+    }
+
+    impl<'a> InternalEvent for SqsMessageProcessingSucceeded<'a> {
+        fn emit_logs(&self) {
+            trace!(message = "Processed SQS message succeededly.", %self.message_id);
+        }
+
+        fn emit_metrics(&self) {
+            counter!("sqs_message_processing_succeeded_total", 1);
+        }
+    }
+
+    #[derive(Debug)]
+    pub(crate) struct SqsMessageProcessingFailed<'a> {
+        pub message_id: &'a str,
+        pub error: &'a ProcessingError,
+    }
+
+    impl<'a> InternalEvent for SqsMessageProcessingFailed<'a> {
+        fn emit_logs(&self) {
+            warn!(message = "Failed to process SQS.", %self.message_id, %self.error);
+        }
+
+        fn emit_metrics(&self) {
+            counter!("sqs_message_processing_failed_total", 1);
+        }
+    }
+
+    #[derive(Debug)]
+    pub(crate) struct SqsMessageDeleteSucceeded<'a> {
+        pub message_id: &'a str,
+    }
+
+    impl<'a> InternalEvent for SqsMessageDeleteSucceeded<'a> {
+        fn emit_logs(&self) {
+            trace!(message = "Deleted SQS message.", %self.message_id);
+        }
+
+        fn emit_metrics(&self) {
+            counter!("sqs_message_delete_succeeded_total", 1);
+        }
+    }
+
+    #[derive(Debug)]
+    pub(crate) struct SqsMessageDeleteFailed<'a> {
+        pub message_id: &'a str,
+        pub error: &'a RusotoError<DeleteMessageError>,
+    }
+
+    impl<'a> InternalEvent for SqsMessageDeleteFailed<'a> {
+        fn emit_logs(&self) {
+            warn!(message = "Deletion of SQS message failed", %self.message_id, %self.error);
+        }
+
+        fn emit_metrics(&self) {
+            counter!("sqs_message_delete_failed_total", 1);
+        }
+    }
+
+    #[derive(Debug)]
+    pub(crate) struct SqsS3EventRecordIgnoredInvalidEvent<'a> {
+        pub bucket: &'a str,
+        pub key: &'a str,
+        pub kind: &'a str,
+        pub name: &'a str,
+    }
+
+    impl<'a> InternalEvent for SqsS3EventRecordIgnoredInvalidEvent<'a> {
+        fn emit_logs(&self) {
+            warn!(message = "Ignored S3 record as event kind that was not ObjectCreated.", %self.bucket, %self.key, %self.kind, %self.name);
```

Comment:
This warning seems fairly technical. I can't quite decipher what this is trying to tell. What does it mean for an "event kind" to not be "ObjectCreated"?

(I'm not only asking for myself, but also as a user, to perhaps guide us towards a better error message)

Reply:
:+1: I'll clarify. This is basically intending to convey that it won't processing other object events (like object deletions). The bucket notifications should only be configured to forward `ObjectCreated` events, but this exists in case they accidentally configure it to send other types too.